rotkehlchen/db/dbhandler.py

```diff
@@ -2338,6 +2350,101 @@ def get_exchange_credentials_extras(self, name: str, location: Location) -> Dict
 
         return extras
 
+    def add_binance_pairs(self, name: str, pairs: List[str], location: Location) -> None:
+        separator = BINANCE_SETTING_SEPARATOR
+        cursor = self.conn.cursor()
+        # retrieve previous information
+        result = cursor.execute(
+            'SELECT setting_value FROM user_credentials_mappings WHERE credential_name=? AND '
+            'credential_location=? AND setting_name=?',
+            (name, location.serialize_for_db(), BINANCE_MARKETS_KEY),
+        )
+        data = result.fetchone()
+        if data:
+            data = data[0]
+            data = separator.join(
+                set(data.split(separator) + pairs),
+            )
+            cursor.execute(
+                'UPDATE user_credentials_mappings '
+                'SET setting_value=? WHERE credential_name=? AND '
+                'credential_location=? AND setting_name=?',
+                (str(data), name, location.serialize_for_db(), BINANCE_MARKETS_KEY),  # noqa: E501
+            )
+        else:
+            data = separator.join(pairs)
+            cursor.execute(
+                'INSERT OR REPLACE INTO user_credentials_mappings '
+                '(credential_name, credential_location, setting_name, setting_value) '
+                'VALUES (?, ?, ?, ?)',
+                (
+                    name,
+                    location.serialize_for_db(),
+                    BINANCE_MARKETS_KEY,
+                    str(data),
+                ),
+            )
+        self.conn.commit()
+        self.update_last_write()
+
+    def set_binance_pairs(self, name: str, pairs: List[str], location: Location) -> None:
```

Comment:
I don't know which of the 2 functions you want to use but why not just use json? Just stringify the json list and deserialize the string to json when reading. json dumps and loads

it would simplify all these functions you wrote here